### PR TITLE
Add location data to Minecart break results

### LINKED ISSUE #60: Add locations to cart breaks

### PR DESCRIPTION
**Pull request by waffle-stomper**
Closes #60

I tried to sign the Contributor License Agreement, but I get a "We're sorry, but something went wrong" error.

### CODE REVIEW

**Comment by Maxopoly** [comment]
Yeah, I think the service we previously used for the CLA no longer exists. Did you test this locally?

**Comment by Maxopoly** [comment]
Change looks good overall, but this would create a weird edge case, where playernames wouldn't show up properly if the player had the same name as any material. Additionally to checking whether the material is not null, could you also check that the action does not equal KILL?

**Comment by waffle-stomper** [comment]
>this would create a weird edge case, where playernames wouldn't show up properly if the player had the same name as any material

Oh wow, I can't believe I missed that. Thanks!
I've added the extra check to fix it.

>Did you test this locally?

I did. I've also tested the new version with a player named 'MINECART' and everything seemed to work as it's supposed to.

**Comment by Maxopoly** [comment]
Ok nice, thanks for fixing.

File: src/com/untamedears/JukeAlert/storage/JukeAlertLogger.java [hunk 1]
```diff
@@ -1359,6 +1359,14 @@ public String createInfoString(ResultSet set, boolean isGroup) {
                     break;
                 case 3:
                     actionText = victim;
+                    // Add location data (if possible)
+                    Material victim_material = Material.matchMaterial(victim);
+                    if (victim_material != null && action != LoggedAction.KILL){
+                        int victim_id = victim_material.getId();
+                        if (victim_id >= 0){
+                            actionText = String.format("%d [%d %d %d]", victim_id, x, y, z);
```
Comment (ProgrammerDan) [inline src/com/untamedears/JukeAlert/storage/JukeAlertLogger.java]: Are we using the ordinal/ material ID here instead of the `victim` longform name to save space?

Comment (Maxopoly) [inline src/com/untamedears/JukeAlert/storage/JukeAlertLogger.java]: Yes, same as broken blocks

Comment (ProgrammerDan) [inline src/com/untamedears/JukeAlert/storage/JukeAlertLogger.java]: Hmm. Like blocks, is this ID added to the list of IDs at the top of the info display?

Comment (Maxopoly) [inline src/com/untamedears/JukeAlert/storage/JukeAlertLogger.java]: Probably not, I didnt even remember there was such a list at the top

**Comment by waffle-stomper** [comment]
>Are we using the ordinal/ material ID here instead of the victim longform name to save space?

I did that to maintain compatibility with [SendSnitchInfo](https://github.com/Civcraft/JukeAlert/blob/master/src/com/untamedears/JukeAlert/chat/SendSnitchInfo.java#L36-L42).

>Like blocks, is this ID added to the list of IDs at the top of the info display?

Yep!

**Comment by Maxopoly** [comment]
Ah okay, nice

**Comment by ProgrammerDan** [comment]
omg, I just had a microstroke.

Great work, been trying to trace that out.

Also, that's terrible (but totally not your fault!). If you get ambitious some day, come back here and rewrite this spaghetti into something that isn't, hehe. Organically grown code is fun but unique!

**Comment by waffle-stomper** [comment]
Thanks for the advice! I have very little formal computer science training so I appreciate any tips.

Once I'm more confident, I might have a go at rewriting that section to remove the string parsing.

**Comment by ProgrammerDan** [comment]
That'd be awesome! It's a bit more complex of a change but it'd be a great way to "get in deep" so to speak.
